lib/cartopy/mpl/style.py

```diff
@@ -0,0 +1,112 @@
+# (C) British Crown Copyright 2018, Met Office
+#
+# This file is part of cartopy.
+#
+# cartopy is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cartopy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with cartopy.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+
+"""
+Handles matplotlib styling in a single consistent place.
+
+"""
+import warnings
+
+import six
+
+
+# Define the matplotlib style aliases that cartopy can expand.
+# Note: This should not contain the plural aliases
+# (e.g. linewidths -> linewidth).
+# This is an intended duplication of
+# https://github.com/matplotlib/matplotlib/blob/\
+#   2d2dab511d22b6cc9c812cfbcca6df3f9bf3094a/lib/matplotlib/patches.py#L20-L26
+# Duplication intended to simplify readability, given the small number of
+# aliases.
+_ALIASES = {
+    'lw': 'linewidth',
+    'ls': 'linestyle',
+    'fc': 'facecolor',
+    'ec': 'edgecolor',
+}
+
+
+def merge(*style_dicts):
+    """
+    Merge together multiple matplotlib style dictionaries in a predictable way
+
+    The approach taken is:
+
+        For each style:
+            * Expand aliases, such as "lw" -> "linewidth", but always prefer
+              the full form if over-specified (i.e. lw AND linewidth
+              are both set)
+            * "color" overwrites "facecolor" and "edgecolor" (as per
+              matplotlib), UNLESS facecolor == "never", which will be expanded
+              at finalization to 'none'
+
+    >>> merge({"lw": 1, "edgecolor": "black", "facecolor": "never"},
+    ...       {"linewidth": 2, "color": "gray"})
+    {'edgecolor': 'gray', 'facecolor': 'never', 'linewidth': 2}
```

Comment:
Can't put `dict`s in doctests; they might change order.

Reply:
Another cut for #1186. 😄